JS/Lesson17/savas/script.js

```diff
@@ -2,97 +2,233 @@
+    sumWords += " " + stringE3[i];
+  }
+  return sumWords;
+}
+
+console.log(`Exersise 3: ${sumStrings(stringE3)}`);
 
 console.log('----------------------------');
 // 4. Write a function that takes an array of integers and returns a new array with odd numbers only.
 // Input: [1, 2, 3, 4, 7, 10]
 // Output: [1, 3, 7]
 let arrayEx = [1, 2, 3, 4, 7, 10];
+function findOddNumbers(num) {
+  let newOddArray = [];
+  for (let i = 0; i < num.length; i++) {
+    if (num[i] % 2 !== 0) {
+    newOddArray.push(num[i]);
+    }
+  }
+  return newOddArray; 
+}
+
+console.log("Exercise 4.1: ", findOddNumbers(arrayEx));
+console.log("Exercise 4.2: ", findOddNumbers([1, 2, 3, 4, 7, 9, 10, 11, 14, 15]));
 
 console.log('----------------------------');
 // 5. Write a function that takes an array of integers and returns a new array with even numbers only.
 // Input: [1, 2, 3, 4, 7, 10]
 // Output: [1, 2, 4, 10]
 
+function findEvenNumbers(num) {
+  let newEvenArray = [];
+  for (let i = 0; i < num.length; i++) {
+  if (num[i] % 2 === 0) {
+    newEvenArray.push(num[i]);
+    }
+  }
+  return newEvenArray; 
+}
+
+console.log("Exercise 5: ", findEvenNumbers(arrayEx));
+
 console.log('----------------------------');
 // 6. Write a function that takes an array and a value, and returns a new array with all occurrences of the value removed.
 // Input: ([1, 2, 3, 4, 2], 2)
 // Output: [1, 3, 4]
 
+function removeNumber(num, removeNum) {
+  let remainArray = [];
+  for (let i = 0; i < num.length; i++) {
+    if (num[i] !== removeNum) {
+      remainArray.push(num[i]);
+    }
+  }
+  return remainArray
+}
+
+console.log("Exercise 6: ", removeNumber([1, 2, 3, 4, 2], 2));
+
 console.log('----------------------------');
 // 7. Write a function that takes an object and returns the keys as an array.
 // Input: {name: "John", age: 30, city: "New York"}
 // Output: ["name", "age", "city"]
 
+function objectKeys(givenObjects) {
+  return Object.keys(givenObjects);
+}
+
+console.log("Exercise 7: ", objectKeys({name: "John", age: 30, city: "New York"}));
+
 console.log('----------------------------');
 // 8. Write a function that takes an object and returns the values as an array.
 // Input: {name: "John", age: 30, city: "New York"}
 // Output: ["John", 30, "New York"]
 
+function objectValues(givenObjects) {
+  return Object.values(givenObjects);
+}
+
+console.log("Exercise 8: ", objectValues({name: "John", age: 30, city: "New York"}));
+
 console.log('----------------------------');
 // 9. Write a function that takes an object and a key, and returns true if the key exists in the object, otherwise false.
 // Input: ({name: "John", age: 30, city: "New York"}, "age")
 // Output: true
 // Input: ({name: "John", age: 30, city: "New York"}, "surname")
 // Output: false
 
+function isKeyHere(givenObjects, givenKeys) {
+    if (typeof givenObjects !== "object") {
+      console.log("Provide value is not an object");
+    } else {
+      if (givenKeys in givenObjects) {
```

Comment:
There's an easier way to check it: `object.hasOwnProperty(testKey);`

Reply:
Is that ok for now?